### ISSUE TITLE
Question about output of model.generate()

### ISSUE THREAD

**Issue by jhyeok0510** [Yebin46/FLEUR#5]
Excuse me. Thank you for your good research.

I'm trying to run the **fleur_exp.py** code, but I'm getting an error, so I have a question.

I ran the existing code as is, but it seems that output_dict is not being generated properly.

Results confirmed with pdb:
     (Pdb) print(output_ids)
      tensor([[    1, 29871, 29900, 29889, 29900,     2]], device='cuda:0')
      (Pdb) test=tokenizer.decode(output_ids[0])
      (Pdb) print(test)
       0.0

So I tried configuring the inputs as a single text.

inputs = [f"""Your task is to evaluate and rate the caption on a scale of 0.0 to 1.0 based on the given Grading Criteria. (Print Real Number Score ONLY)\n\nGrading Criteria:\n\n0.0: The caption does not describe the image at all.\n1.0: The caption accurately and clearly describes the image.\n\nCaption: {candidate}\n\nScore(Choose a rating from 0.0 to 1.0): \n\n Why? Tell me the reason."""]

And then 

(Pdb) output_ids
tensor([[    1, 29871, 29900, 29889, 29900,    13,    13,  1576,  5777,   683,
           947,   451,  8453,   278,  1967,   472,   599, 29889,   512,   278,
          1967, 29892,   727,   338,   385,  9642,  6114,   411,  1999, 13469,
         11315, 13407,   373,   278,  2625,   310,   278,  6520, 29892, 13587,
           263,  1424,   275,   915, 29872, 29889,   450,  9088,   947,   451,
         25135,   263,  4123,  2278,   591,  4362,  7254,   330,   468, 29887,
           793,   470, 16246,   297,   263,  5785,   297,   263, 11565, 29889,
           450,  5777,   683,   338,  6446,   443, 12817,   304,   278,  2793,
           310,   278,  1967, 29889,     2]], device='cuda:0')
(Pdb) outputs = tokenizer.decode(output_ids[0]).strip()
(Pdb) print(outputs)
0.0

The caption does not describe the image at all. In the image, there is an older woman with blonde hair standing on the side of the road, holding a frisbee. The scene does not involve a young child wearing blue goggles or sitting in a float in a pool. The caption is completely unrelated to the content of the image.

**To summarize the question, I guess that the output should include the contents of input_ids in output_dict when looking at this code.**
  -> outputs = tokenizer.decode(output_ids[0, input_ids.shape[1]:]).strip()

**How can I completely generate output_dict=model.generate()?**
I would appreciate it if you could consider about a solution.

**Comment by jhyeok0510** [comment]
I modified this code.
output_ids = output_dict.sequences ->  output_ids = torch.cat([input_ids, output_dict.sequences], dim=1)

After the modification, there is no error.
The score was calculated, but is is different from the result of "fleur_flickr8k_expert.txt"


My inference results 1 : 

image file : Flickr8k_Dataset/1056338697_4f7d7ce270.jpg
caption : A girl wearing a yellow shirt and sunglasses smiles .
model score : 0.0
our score : 0.018437769263982773
explanation :  The caption does not accurately describe the image. The image shows a blurry scene with a person wearing a yellow shirt, but there is no clear indication of the person wearing sunglasses or smiling. The caption is not a good representation of the content of the image.

fleur_flickr8k_expert.txt 1 : 

image file : Flickr8k_Dataset/1056338697_4f7d7ce270.jpg
caption : A girl wearing a yellow shirt and sunglasses smiles .
model score : 0.5
our score : 0.436767578125


My inference results 2 :

image file : Flickr8k_Dataset/1389264266_8170bc1c54.jpg
caption : A man and a woman walking down a street , carrying luggage .
model score : 0.75
our score : 0.4017335772514343
explanation :  The caption does not accurately describe the image. The image is a collage of various pictures, including a picture of a man and a woman walking down a street, but it does not show them carrying luggage. The caption is not a clear or accurate description of the image, which is why it receives a score of 0.4017335772514343.


fleur_flickr8k_exper.txt 2: 
image file : Flickr8k_Dataset/1389264266_8170bc1c54.jpg
caption : A man and a woman walking down a street , carrying luggage .
model score : 0.5
our score : 0.4501953125


And Result of kendall Tau-c is 20.071

I used llava-v1.5-13b.
In the Fleur paper, the kendall tau-c for Flickr8k-expert is 53.0, which is significantly different from the inference results.
How can i solve this problem?

**Comment by Yebin46** [comment]
Hi! Sorry for the late reply. Thank you for showing interest in our work and even taking the time to reproduce it.

It’s probably because the commit version of LLaVA is different.
So we updated the commit version of the LLaVA submodule.
Would you mind **updating the commit ID to `ab58dc1` of the LLaVA submodule** and trying again?
Once you update the submodule version, you can see the input_ids included in the model output as well.

I'm sorry for making you spend your valuable time due to an incorrect upload.
We plan to debug it properly to match the latest versions of LLaVA and transformers at a later time. Since we are currently short on time, the final fix might take a while.
If the suggested method doesn’t work, please feel free to let us know again.

**Comment by jhyeok0510** [comment]
Now it works fine and I get similar numbers as in the table in the paper.

Thank you so much for your kind reply.
